pkg/pillar/netmonitor/linux.go

```diff
@@ -557,6 +557,7 @@ func (m *LinuxNetworkMonitor) linkSubscribe(doneChan chan struct{}) chan netlink
 	}
 	linkOpts := netlink.LinkSubscribeOptions{
 		ErrorCallback: linkErrFunc,
+		ListExisting:  true, // XXX: currently required by zedrouter (later will be removed)
```

Comment:
How do you plan to remove the listExisting?
Can't use a get+subscription since the order would not be defined between what you retrieve with a get and when you see any additions or deletings from the subscription.

Reply:
Since we have very small routing tables (only getting routes from DHCP or static config, no dynamic routing protocols), I would prefer to fully reconcile a routing table on change (reread the current state, update the intended state, run reconcile). In other words, use these notifications only to detect that config is out-of-sync. It should be more robust at the cost of more netlink calls made overall.
But I may rethink this once I get to it, so for now keeping there the `XXX` comment.

Reply:
Sure, such an approach can avoid race conditions as long as you notification generated after you start the read of the current causes the code to redo the full read. I don't know if you also need to make sure you discard the results of that first read - whether the set of routes you read could be inconsistent. By that I mean that a read could have e.g., a default route with a nexthop of 192.168.17.1 but you have no route for 192.168.17.1 in the set of routes you read (because that interface/route was deleted just after netlink read the default route.)